dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -182,7 +183,6 @@ def load_user
     # TODO: This should use cancan/authorize.
     if user.student_of?(current_user)
       @user = user
-      @user_level = @user.user_level_for(@script_level)
```

Comment:
`@user_level` is only used in rendering the `levels/teacher` partial, when the teacher is viewing a student's solution. `load_level_source` is always called before rendering `levels/show`, so it's safe to set `@user_level` there instead, and the section I moved it to is for handling the "teacher viewing student's solution" case as well.

I moved it because I now need to determine which level to render before setting `@user_level`.